apps/omg/test/omg/state/fee_test.exs

```diff
@@ -0,0 +1,50 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.State.FeeTest do
+  @moduledoc false
+
+  use ExUnitFixtures
+  use ExUnit.Case, async: true
+
+  alias OMG.State.Transaction
+
+  @eth OMG.Eth.RootChain.eth_pseudo_address()
+
+  @tag fixtures: [:alice]
+  test "can be encoded to binary form and back", %{alice: owner} do
+    fee_tx = Transaction.Fee.new(1000, {owner.addr, @eth, 1551})
+
+    rlp_form = Transaction.raw_txbytes(fee_tx)
+    assert fee_tx == Transaction.decode!(rlp_form)
+  end
+
+  @tag fixtures: [:alice]
+  test "hash can be computed with protocol implementation", %{alice: owner} do
+    fee_tx = Transaction.Fee.new(1000, {owner.addr, @eth, 1551})
+    fee_txhash = Transaction.raw_txhash(fee_tx)
+    assert <<_::256>> = fee_txhash
+
+    assert Transaction.raw_txhash(Transaction.Fee.new(1000, {owner.addr, @eth, 1551})) == fee_txhash
+    assert Transaction.raw_txhash(Transaction.Fee.new(1001, {owner.addr, @eth, 1551})) != fee_txhash
+  end
+
+  @tag fixtures: [:alice]
+  test "cannot be recovered in payment scenario", %{alice: owner} do
+    fee_tx = Transaction.Fee.new(1000, {owner.addr, @eth, 1551})
+    tx_rlp = Transaction.Signed.encode(%Transaction.Signed{raw_tx: fee_tx, sigs: []})
+
+    assert {:error, :transaction_not_transfer_funds} = Transaction.Recovered.recover_from(tx_rlp)
```

Comment:
oh, but watchers are going to do this! All transactions must be recoverable, no? I think the interfaces like `valid?` and `can_apply?` should be properly implemented

Reply:
maybe I'm missing some scope-of-task context here, but I'd expect `OMG.State.Core`-level tests to be there (spendability of the fee output and others). If that is in some follow-up ticket's scope, then I'd vote for doing the `recover_from` functionality more resembling the final form

Reply:
Yes, it's scope issue - original omisego/feat-fee#27 was splitted and this task brings just structures that feature will be implemented on. Actual behavior will be refined in the next tasks